src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -249,6 +275,20 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
             }
         }
 
+        private static void AppendDeletedProperties(dynamic resource, ODataDeletedResource deletedResource, bool isUntyped)
+        {
+            if (isUntyped)
+            {
+                resource.Id = deletedResource.Id.ToString();
+            }
+            else
+            {
+                resource.Id = deletedResource.Id;
+            }
+
+            resource.Reason = deletedResource.Reason.Value;
```

Comment:
`Reason` is nullable. `deletedResource.Reason.Value` will throw an `InvalidOperationException` if `deletedResource.Reason` has a value of `null`.
Is there any guarantee that it'll have a value here or should we be accessing `Value` conditionally?